src/lavinmq/queue/message_store.cr

```diff
@@ -305,10 +305,6 @@ module LavinMQ
               @size += 1
             end
           rescue ex : IO::EOFError
-            if mfile.pos < mfile.size
-              Log.warn { "Truncating #{mfile.path} from #{mfile.size} to #{mfile.pos}" }
-              mfile.truncate(mfile.pos)
```

Comment:
`@size` still needs to be updated

Reply:
> `@size` still needs to be updated

updated so we do `mfile.resize if ts.zero?` but still truncating if EOF or message doesn't fit

Reply:
gr8!